### PR TITLE
automatically picked filetype

### PR DESCRIPTION
**Pull request by asterized**

### CODE REVIEW

**Comment by asterized** [comment]
resolves #11

File: umm/file/__init__.py [hunk 1]
```diff
@@ -95,29 +96,40 @@ def validate(
...
+    file_type: Annotated[filetypes, typer.Option("--type", "-t")] = None,
 ) -> None:
     """Convert UMM to other filetypes"""
     output_file = pathlib.Path(output_file).resolve()
     umm = get_umm(input_file)
     umm_version = umm["info"]["version"]
-    match file_type.value:
+
+    if file_type is None:
+        if not (file_type := output_file.suffix[1:]):
```
Comment (Gorian) [inline umm/file/__init__.py:86]: we don't want to use the file extension here - let's write some code to update the mapping_json and store the file extension with the mapping, and instead just pull the names from the mappings file. Much easier

File: umm/file/__init__.py [hunk 2]
```diff
@@ -95,29 +96,40 @@ def validate(
...
             template_file = (
                 importlib.resources.files("umm") /
                 "assets" /
                 "templates" /
-                file_type.value /
+                file_type /
                 f"{template_version}.template"
             )
-            print(_version_map("opf", umm_version))
+            print(_version_map(file_type, umm_version))
```
Comment (Gorian) [inline umm/file/__init__.py]: This is just testing code anyway, so it doesn't matter, lol

File: umm/file/__init__.py [hunk 3]
```diff
@@ -25,40 +25,18 @@
 # endregion
 
 cmd = typer.Typer()
+mapping_file = importlib.resources.files("umm") / "assets" / "mapping.json"
```
Comment (Gorian) [inline umm/file/__init__.py:28]: This should probably be moved to a constant, but not a big deal

File: umm/file/__init__.py [hunk 4]
```diff
@@ -123,18 +105,19 @@ def convert(
...
-                importlib.resources.files("umm") /
-                "assets" /
-                "templates" /
-                file_type /
-                f"{template_version}.template"
+                importlib.resources.files("umm")
+                / "assets"
+                / "templates"
+                / file_type
+                / f"{template_version}.template"
```
Comment (Gorian) [inline umm/file/__init__.py:112]: You keep switching these back and forth, lol. Probably unnecessary

File: umm/file/__init__.py [hunk 5]
```diff
@@ -123,18 +105,19 @@ def convert(
...
             )
-            print(_version_map(file_type, umm_version))
             with open(template_file, mode="r", encoding="utf-8") as file:
                 template = Template(file.read())
             with open(output_file, "w", encoding="utf-8") as file:
                 file.write(template.render(umm=umm))
 
+            print(f"Successfully converted {input_file} to {output_file}.")
```
Comment (Gorian) [inline umm/file/__init__.py:119]: We should probably think about things like this for a production release - you may not want the application to output text by default for the purposes of automation 🤔

**Review by Gorian — APPROVED** [review]

**Pull request merged**